src/main/java/duke/MeowBot.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+import java.io.IOException;
+
+import duke.command.Command;
+import duke.helper.Parser;
+import duke.helper.Storage;
+import duke.helper.Ui;
+import duke.task.DukeException;
+import duke.task.TaskList;
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.shape.Circle;
+import javafx.stage.Stage;
+
+
+/**
+ * MeowBot is the main class which runs the iP project in its main function
+ * @since 2023-08-30
+ */
+
+
+public class MeowBot {
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+    private final String FILE_NAME;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Represents the 2 possible inputs, that is either User Dialog or response from MeowBot
+     */
+
+    public enum dialog {
+        USER,
+        MEOWBOT
+
+    }
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    /**
+     * @param file indicates the location of the file where data is stored
+     * @throws DukeException when generating the tasks back from the data file
+     * @throws IOException when the data file cannot be found
+     */
+
```

Comment:
Unnecessary line here between javadoc comment and method. Consider removing this to keep thinks more compact!